### PR TITLE
Do not crash timeline with out of spec bytecode transition

### PR DESCRIPTION
**Pull request by jonaias**
OK to merge.

The attached project ( https://share.haiku.ai/u/taylor/booptroop/fork ) had an out of spec transition:
``` javascript
        "translation.x": {
          "0": { value: 329.5, edited: true },
          "400": { value: 315.5, edited: true, curve: "linear" },
          "933": { value: 390, edited: true, curve: "linear" }
        },
```
When TransitionBody was created for the second transition, it didn't have next keyframe, so it was crashing timeline. I've tested (Hey @roperzh can you do some testing as you know how to achieve timeline edge cases?),  and the user can edit the element transition without crash after this commit. 

Question: Should we warn the user about the non conformant bytecode?

Completed checkin tasks:
- [x] Did manual testing of interrelated functionality
- [x] Updated `changelog/public/latest.json`

### CODE REVIEW

File: changelog/public/latest.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
       "Snap lines now have a consistent thickness regardless of the artboard zoom level.",
       "Export options are now disabled outside of a project.",
       "User Menu -> Your Profile now points to the correct location.",
-      "Resolved a breaking change in Lottie Android 2.6.0 causing crashes."
+      "Resolved a breaking change in Lottie Android 2.6.0 causing crashes.",
+      "Improve project forking robustness with out of spec transitions"
```
Comment (stristr) [inline changelog/public/latest.json]: I might reword this as "Fixed a crash caused by forking certain projects." to avoid overwhelming users with TMI.

**Review by stristr — APPROVED** [review]
I know you asked for @roperzh, but LGTM :smile:

File: changelog/public/latest.json (hunk 1 above)
Comment (jonaias) [reply changelog/public/latest.json]: Much better :+1:

**Review by roperzh — APPROVED** [review]
LGTM! Just a note, some time a go we had the [same crash](https://the-haiku-team.slack.com/archives/C6XKR303S/p1535066067000100) and @stristr solved the underlying issue (which was related to publishing at that time) instead of doing this. Is this a symptom of something else going on or should we be doing this anyway?

**Comment by stristr** [comment]
1. Definitely in favor of doing this anyway (should have done it the last time it was broken). Generally, unchecked deep property access through nullable/user-generated data fields is an accident waiting to happen.
2. This particular problem seems to be a consequence of assuming if a timeline keyframe has a "curve", it also has a "next keyframe". Not sure how this might come to _not_ be the case, but I can think of two ways. The first is putting three keyframes in one timeline row, connecting them with two tweens, and then deleting the final keyframe. The second is typing it in bytecode. Because the second always might happen even if we fix the first, I think we still want/need this fix.

**Comment by roperzh** [comment]
✊ 🔥✊ merge, merge merge! ✊🔥 ✊
